src/batched/KokkosBatched_SolveUTV_TeamVector_Internal.hpp

```diff
@@ -142,39 +143,45 @@ namespace KokkosBatched {
 		   one,
 		   T, ts0, ts1,
 		   W, ws0, ws1);
+	member.team_barrier();
 	
 	/// X = V^T W
 	TeamVectorGemmInternal<Algo::Gemm::Unblocked>
 	  ::invoke(member,
-		   m, nrhs, matrix_rank, 
+		   n, nrhs, matrix_rank, 
 		   one,
 		   V, vs1, vs0,
 		   W, ws0, ws1,
 		   zero,
 		   X, xs0, xs1);
+	member.team_barrier();
       } else {
+	/// W = U^T B
 	TeamVectorGemmInternal<Algo::Gemm::Unblocked>
 	  ::invoke(member,
-		   m, nrhs, matrix_rank, 
+		   matrix_rank, nrhs, m, 
 		   one,
 		   U, us1, us0,
 		   B, bs0, bs1,
 		   zero,
 		   X, xs0, xs1);
+    member.team_barrier();
 
+	/// X = T^{-1} X
 	TeamVectorTrsmInternalLeftUpper<Algo::Trsm::Unblocked>
 	  ::invoke(member,
 		   false,
 		   matrix_rank, nrhs,
 		   one,
 		   T, ts0, ts1,
 		   X, xs0, xs1);		
+	member.team_barrier();
```

Comment:
possibly not needed since Trsm should always end on a diagonal matrix entry, the calculation of which has a team_barrier(). Maybe you could verify?